sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOTest.java

```diff
@@ -427,15 +441,15 @@ static KafkaIO.Read<Integer, Long> mkKafkaReadTransform(
       reader = reader.withTimestampFn(timestampFn);
     }
 
-    if (redistribute) {
+    if (redistribute != null && redistribute) {
+      reader = reader.withRedistribute();
+      reader = reader.withAllowDuplicates(withAllowDuplicates);
```

Comment:
seems like this should have null check too?

Reply:
It is checked and overridden at L436-L438 above.